### DISCUSSION TITLE
How can I set base url for react-native-web?

### DISCUSSION THREAD

**Discussion by WayneWang98** [react-navigation/react-navigation#11403] [Q&A]
I migrated react-navigation from **v4 to v6**. It seems that react-navigation v6 doesn't support HashHistory, instead using WebHistory.
It works fine on local development environment, but will give us a 404 error when user refresh the web sub-page on production environment.(e.g.: https://example.com/Main/AuthNavigator/Welcome)
I found that it was caused because the server side didn't have the resource which matched to the example URL. So I want to use **Nginx** to rewrite the resource to index.html, but we hope all the requests send to URL must have a base URL '/web-portal' due to some limitation of our Nginx configurations.

**Current Situation:**
There are two stacks which named Main and Modal in our app.  The URL looks like this: https://example.com/Main/AuthNavigator/Welcome;
https://example.com/Main/AuthNavigator/Login

**Expectation:**
I hope that no matter how the route is changed, the common prefix '/web-portal' is always maintained in the route.
https://example.com/web-portal/Main/AuthNavigator/Welcome;
https://example.com/web-portal/Main/AuthNavigator/Login;

What should I do to implement this feature? Or, can we use HashHistory mode in react-navigation v6? If we can use HashHistory mode, this 404 issue may not happen.
Thanks.

**Comment by JPStrydom** [comment]
We've been looking for a solution to a similar problem. 

Basically we want to host our app under `https://system.com` and our test environment under `https://system.com/test`. Is there a way to specify the base URL for a React Navigation react-native-web web app?

There are these two Stack Overflow issues that relate to this question, but neither have proper answers. 
1. [https://stackoverflow.com/questions/71597798/how-to-specify-base-url-path-for-react-navigation-web](https://stackoverflow.com/questions/71597798/how-to-specify-base-url-path-for-react-navigation-web)
2. [https://stackoverflow.com/questions/66728161/integrate-react-native-web-into-django-how-to-add-a-prefix-to-all-urls](https://stackoverflow.com/questions/66728161/integrate-react-native-web-into-django-how-to-add-a-prefix-to-all-urls)

If anyone has an idea of how we can do this it's be greatly appreciated.

**Reply by satya164** [reply]
It's possible in React Navigation 7

https://reactnavigation.org/docs/7.x/configuring-links#apps-under-subpaths

However, I'd recommend using a subdomain instead of a subpath. Production and test environments should behave similarly.

**Reply by JPStrydom** [reply]
> It's possible in React Navigation 7
> 
> https://reactnavigation.org/docs/7.x/configuring-links#apps-under-subpaths
> 
> However, I'd recommend using a subdomain instead of a subpath. Production and test environments should behave similarly.

Many thanks for this! Will give it a try.

We'd also prefer a sub domain, but the client only has one domain available for us to deploy to.
